### PR TITLE
Various minor fixes

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary
Various minor fixes

## Testing
local sim

### CODE REVIEW

**Review by anchao — APPROVED** [review]

File: drivers/1wire/ds28e17.c [hunk 1]
```diff
@@ -537,11 +537,12 @@ static int ds_i2c_setfrequency(FAR struct ds_i2c_inst_s *inst,
         break;
 
       default:
-        i2cerr("ERROR: bad I2C freq %lu\n", frequency);
+        i2cerr("ERROR: bad I2C freq %" PRIx32 "\n", frequency);
```
Comment (hartmannathan) [inline drivers/1wire/ds28e17.c:540]: Note: This changes the output from base-10 to base-16 (which is okay as long as it's intentional).

File: drivers/1wire/ds28e17.c [hunk 2]
```diff
@@ -537,11 +537,12 @@ static int ds_i2c_setfrequency(FAR struct ds_i2c_inst_s *inst,
...
         return -EINVAL;
     }
 
-  i2cinfo("Changing I2C freq %lu -> %lu\n", inst->frequency, frequency);
+  i2cinfo("Changing I2C freq %" PRIx32 " -> %" PRIx32 "\n",
```
Comment (hartmannathan) [inline drivers/1wire/ds28e17.c:544]: Ditto.

File: drivers/motor/stepper.c [hunk 3]
```diff
@@ -266,7 +266,7 @@ static int stepper_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
         {
           lower->status.position = (int32_t)arg;
           ret = 0;
-          stpinfo("STEPIOC_SET_CURRENT_POS: new position is %ld\n",
+          stpinfo("STEPIOC_SET_CURRENT_POS: new position is %" PRIx32 "\n",
```
Comment (hartmannathan) [inline drivers/motor/stepper.c:269]: Ditto.

File: drivers/1wire/ds28e17.c (hunk 1 above)
Comment (raiden00pl) [reply drivers/1wire/ds28e17.c]: changed back to base-10

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**